packages/ckeditor5-engine/src/view/matcher.ts

```diff
@@ -461,15 +465,13 @@ function matchAttributes(
 
 	// `style` and `class` attribute keys are deprecated. Only allow them in object pattern
 	// for backward compatibility.
-	if ( isPlainObject( patterns ) ) {
-		if ( ( patterns as any ).style !== undefined ) {
-			// Documented at the end of matcher.js.
-			logWarning( 'matcher-pattern-deprecated-attributes-style-key', patterns as any );
-		}
-		if ( ( patterns as any ).class !== undefined ) {
-			// Documented at the end of matcher.js.
-			logWarning( 'matcher-pattern-deprecated-attributes-class-key', patterns as any );
-		}
+	if ( ( patterns as any ).style !== undefined ) {
```

Comment:
I'm not 100% sure about removing this `if` statement.

If it's needed, when add the following and remove the `as any` castings:

```ts
	if ( typeof patterns === 'object' && !( patterns instanceof RegExp ) && !Array.isArray( patterns ) ) {
```